enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/APIKeyAuthenticator.java

```diff
@@ -76,113 +76,47 @@ public APIKeyAuthenticator() {
-        if (isAppLevelApiKeySecurityEnabled) {
-            // If API key enabled at Application level via APIM, key can exist in header or query param
-            String apiKeyName = requestContext.getMatchedAPI().
-                    getSecuritySchemeDefinitions().get(APIConstants.API_SECURITY_API_KEY).getName();
-            if (requestContext.getHeaders().containsKey(apiKeyName)) {
-                return requestContext.getHeaders().get(apiKeyName);
-            }
-            if (requestContext.getQueryParameters().containsKey(apiKeyName)) {
-                return requestContext.getQueryParameters().get(apiKeyName);
-            }
-        } else {
-            apiKey = getAPIKey(APIConstants.SWAGGER_API_KEY_AUTH_TYPE_NAME, requestContext, true);
-            if ("".equals(apiKey)) {
-                apiKey = getAPIKey(APIConstants.SWAGGER_API_KEY_AUTH_TYPE_NAME, requestContext, false);
-            }
-        }
-        return apiKey;
-    }
-
-    private static String getAPIKey(String securitySchemeType, RequestContext requestContext, boolean isResourceLevel) {
-        Map<String, List<String>> securitySchemesToApply;
-        if (isResourceLevel) {
-            securitySchemesToApply = requestContext.getMatchedResourcePath().getSecuritySchemas();
-        } else {
-            securitySchemesToApply = requestContext.getMatchedAPI().getApiSecurity();
-        }
-
+    private static String getAPIKeyFromRequest(RequestContext requestContext) {
         Map<String, SecuritySchemaConfig> securitySchemaDefinitions = requestContext.getMatchedAPI().
                 getSecuritySchemeDefinitions();
-
-        for (String securitySchemeName : securitySchemesToApply.keySet()) {
-            SecuritySchemaConfig securitySchemaDefinition = securitySchemaDefinitions.get(securitySchemeName);
-
-            // We only need apiKey of the given type
-            if (securitySchemaDefinition != null && securitySchemeType.equalsIgnoreCase(
-                    securitySchemaDefinition.getType())) {
-
-                // If Defined in openAPI definition (when not enabled at APIM App level),
-                // key must exist in specified location
-                if (APIConstants.SWAGGER_API_KEY_IN_HEADER.equalsIgnoreCase(
-                        securitySchemaDefinition.getIn())) {
-                    if (requestContext.getHeaders().containsKey(securitySchemaDefinition.getName())) {
-                        return requestContext.getHeaders().get(securitySchemaDefinition.getName());
-                    }
-                }
-                if (APIConstants.SWAGGER_API_KEY_IN_QUERY.equalsIgnoreCase(
-                        securitySchemaDefinition.getIn())) {
-                    if (requestContext.getQueryParameters().containsKey(securitySchemaDefinition.getName())) {
-                        return requestContext.getQueryParameters().get(securitySchemaDefinition.getName());
+        // check both resource level and api level security
```

Comment:
We must check if the security scheme was originally defined before checking in the request. We need this  https://github.com/wso2/product-microgateway/pull/2501/files#diff-de53b714b118fea8cc93f995cefa65d4edbb54dc569e5dbe8038bbfbf4a23ef3L139-L143